tests/printer.rs

```diff
@@ -491,3 +491,24 @@ fn disable_url_links() -> Result<(), MietteError> {
     assert!(out.contains("oops::my::bad"));
     Ok(())
 }
+
+#[test]
+fn unnamed_snippet_shows_message() {
+    #[derive(Debug, Diagnostic, Error)]
+    #[error("oops!")]
+    #[diagnostic(code(oops::my::bad), help("try doing it better next time?"))]
+    struct MyBad {
+        src: String,
+        #[snippet(src, message("This is the part that broke"))]
+        ctx: SourceSpan,
+    }
+    let src = "source_text_here".to_string();
+    let len = src.len();
+    let err = MyBad {
+        src,
+        ctx: (0, len).into(),
+    };
+    let out = fmt_report(err.into());
+    println!("{}", out);
+    assert!(out.contains("This is the part that broke"));
```

Comment:
Could you please compare against the entire, rendered message. I made an exception for URLs because these tests (intentionally) don't test the escape codes themselves.

Reply:
Yep

Reply:
For another day, this might be a job for https://lib.rs/insta, to do what you've basically been doing manually in `tests/narrative.rs`

Reply:
yes that would be great!